assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -658,57 +675,81 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
-  let height = dimensions?.height ?? 0;
-  let width = dimensions?.width ?? 0;
+  // Get the size of the diagram line map svg, excluding emphasis
+  const [lineDiagramHeight, setLineDiagramHeight] = useState(0);
+  const [lineDiagramWidth, setLineDiagramWidth] = useState(0);
+
+  // Measures line-map svg when the scaleFactor changes, updates state
+  const measuredRef = useCallback(node => {
+    const myObserver = new ResizeObserver(() => {})
+
+    if (node !== null) {
+      myObserver.observe(node)
+      setLineDiagramHeight(node.getBoundingClientRect().height);
+      setLineDiagramWidth(node.getBoundingClientRect().width);
+    }
+    
+    return () => myObserver.disconnect()
+  }, [scaleFactor]);
   
+  // Scale the line-map svg given the available screen width 
   const measureDiagramAndScale = () => {
-    // Get updated dimensions each time this hook runs
-    dimensions = document.getElementById("line-map")?.getBoundingClientRect();
-    height = dimensions?.height ?? 0;
-    width = dimensions?.width ?? 0;
-
-    if (svgHeight != 0 && width && height) {
-      // if scaleFactor has already been applied to the line-map, we need to reverse that
-      const unscaledHeight = height / scaleFactor;
-      const unscaledWidth = width / scaleFactor;
+    if (!isDone && diagramContainerHeight != 0) {
+      // If scaleFactor has already been applied to the line-map, we need to reverse that for calculations
+      const unscaledHeight = lineDiagramHeight / scaleFactor;
+      const unscaledWidth = lineDiagramWidth / scaleFactor;
 
       // First, scale x. Then, check if it needs abbreviating. Then scale y, given the abbreviation
-      let xScaleFactor = 904 / unscaledWidth;
+      let xScaleFactor = fullWidth / unscaledWidth;
       
       const needsAbbreviating = !doAbbreviate &&
-        unscaledHeight * xScaleFactor + getEmphasisHeight(xScaleFactor) > svgHeight;
+        unscaledHeight * xScaleFactor + getEmphasisHeight(xScaleFactor) > diagramContainerHeight;
       if (needsAbbreviating) {
         setDoAbbreviate(true);
         // now scale y, which requires re-running this effect
       } else {
-        const yScaleFactor = (svgHeight - getEmphasisHeight(1)) / unscaledHeight
+        const yScaleFactor = (diagramContainerHeight - getEmphasisHeight(1)) / unscaledHeight
         const factor = Math.min(
           xScaleFactor,
           yScaleFactor
         );
         setScaleFactor(factor);
-        setTimeout(() => {
-          setIsDone(true);
-        }, 200);
+        setIsDone(true);
       }
     }
   }
 
   // When the parent container size changes, or abbreviation setting changes,
-  // re-measure the diagram and scale accordingly
+  // re-measure the diagram and scale accordingly.
   useEffect(() => {
-    setTimeout(() => {
-      measureDiagramAndScale()
-    }, 100)
-  }, [svgHeight, doAbbreviate]);
+    // The isCurrent setting is needed to clean up the unused hook runs / state changes
+    let isCurrent = true
```

Comment:
Oof this is tricky. I think it works only because `isCurrent` is checked within a separate callback, but it does seem sound.